lib/Serialization/Deserialization.cpp

```diff
@@ -5269,16 +5269,10 @@ class TypeDeserializer {
     // FIXME: Check this?
     auto parentTy = MF.getType(parentID);
 
-    SmallVector<Type, 8> genericArgs;
-    for (TypeID ID : rawArgumentIDs) {
-      auto argTy = MF.getTypeChecked(ID);
-      if (!argTy)
-        return argTy.takeError();
-
-      genericArgs.push_back(argTy.get());
-    }
+    // Read the substitutions.
+    const auto subMap = MF.getSubstitutionMap(substitutionsID);
```

Comment:
I think we need to refactor getSubstitutionMap() to propagate failure instead of crashing, then, since the old behavior here was that a failure to deserialize a generic argument would propagate out instead of crashing.

Reply:
Oh, I missed this, thank you! We can use `getSubstitutionMapChecked`; interestingly, other type nodes that deal with substitution maps all use the unchecked variant.

Reply:
Since you're here do you mind fixing those in a separate PR?

Reply:
Sure. It is strange though. Some stuff is checked for and other is not, as if deliberately.